### PR TITLE
add cpp linking test

### PR DESCRIPTION
**Pull request by ryjones**
This adds a linking to cpp test.

This modifies the build action to build the test.

### CODE REVIEW

**Comment by ryjones** [comment]
Hi @aidenfoxivey I moved the code to this PR

**Comment by aidenfoxivey** [comment]
I believe this is a good contender for a "how many PRs does it take an engineering student to complete a 2 file change" joke lol. @ryjones Thank you so much for your help.

File: .github/workflows/basic.yml
Comment (Martyrshot) [inline .github/workflows/basic.yml:94]: Do we need this environment variable? I think we only need the SIG_NAME for this test.

File: .github/workflows/basic.yml
Comment (Martyrshot) [inline .github/workflows/basic.yml:105]:
```suggestion
            -DOQS_MINIMAL_BUILD="SIG_$SIG_NAME" \
```

**Review by Martyrshot — DISMISSED** [review]
This looks good to me! Just a small nit. Thanks for working on this. 😄

File: .github/workflows/basic.yml
Comment (aidenfoxivey) [reply .github/workflows/basic.yml:105]: Yep, this makes sense.

File: .github/workflows/basic.yml
Comment (aidenfoxivey) [reply .github/workflows/basic.yml:94]: No, don't believe we do - I had left it in to be cautious, but it shouldn't be necessary.

**Comment by aidenfoxivey** [comment]
One thing I had considered, was redirecting output from running the binary, but I think it's better to keep the trace on stderr for debugging.

**Review by dstebila — APPROVED** [review]

File: cpp/cpp_linking_example.cpp
Comment (baentsch) [inline cpp/cpp_linking_example.cpp:66]: Hmm -- I'm not sure I understand this: CI seems to _disable_ Dilithium2 and this test basically only runs when Dilithium2 is _enabled_, so what is actually tested in CI @aidenfoxivey @ryjones ?

Comment (ryjones) [reply cpp/cpp_linking_example.cpp:66]: @aidenfoxivey propose a change in a review comment

Comment (baentsch) [reply cpp/cpp_linking_example.cpp:66]: > I don't have access to the repository to make a change right now.

Why can't you do a proper PR from your own repo @aidenfoxivey ? Proceeding with this PR would make @ryjones the author of record -- which simply is wrong on several counts.

If the problem is that [no reviewer/committer presses the "execute CI" button](https://github.com/open-quantum-safe/liboqs/pull/1957#issuecomment-2436069221) on a PR, well, then this project has some serious problem of getting enough attention... 

@aidenfoxivey if in turn you have problems with the [contribution process](https://github.com/open-quantum-safe/liboqs/blob/main/CONTRIBUTING.md) (where actually the way how to run CI also locally is described) please let us know as part of your PR: We want to improve our documentation such that new contributors can easily, well, contribute.

Comment (ryjones) [reply cpp/cpp_linking_example.cpp:66]: @baentsch the issue was that the PR from his repo cannot execute all of the CI due to no access to secrets.

Comment (ryjones) [reply cpp/cpp_linking_example.cpp:66]: @baentsch It isn't obvious, but I did do non-zero work on this PR.

As far as token availability, this is a limitation (and one I agree with) of the GitHub platform. The way to "work around it" is to set up guardrails around jobs (like triggering CircleCI) that require the job to be within the repo, not a forked repo. I planned to do this later in a follow-on PR for multiple jobs in the org, but that is an early next year thing, not a next week thing.

Comment (aidenfoxivey) [reply cpp/cpp_linking_example.cpp:66]: Regarding next steps, I'm fine to make another pull request and move my changes there. In addition, I'm absolutely fine to just have this merged as is. I'll defer to others on the best approach for this.

Comment (baentsch) [reply cpp/cpp_linking_example.cpp:66]: > It isn't obvious, but I did do non-zero work on this PR.

Thanks for letting me know, @ryjones . I indeed didn't see this. So who would you see as the person that should be primarily attributed with this code (and how to properly attribute the other)?

My proposal would be to do it "as usual", i.e., ask @aidenfoxivey to do a PR from his/her repo and you propose your contributions during the PR: That way, both authors are recorded and we see CI run where possible (and trigger the rest). @aidenfoxivey, it would be nice if you'd [locally execute the new test via CI](https://github.com/open-quantum-safe/liboqs/blob/main/CONTRIBUTING.md#running-ci-locally) (also checking that it actually runs :) before the PR. If you encounter problems in following the documented guidance on that, please let us know.

Comment (ryjones) [reply cpp/cpp_linking_example.cpp:66]: @baentsch : @aidenfoxivey did most of the work, I did some of the work. I'm fine with having my attribution removed as Aiden would have made it there eventually.

I agree with having @aidenfoxivey create a new PR. As it's been approved twice, I think it should be merged as-is and allow Aiden to iterate in follow-up PRs.

**Comment by aidenfoxivey** [comment]
Moved my changes over to #1971

**Pull request closed without merging**
